client/app/vps/monitoring/vps-monitoring.controller.js

```diff
@@ -0,0 +1,79 @@
+class VpsMonitoringCtrl {
+    constructor ($q, $stateParams, $translate, CloudMessage, VpsActionService, VpsMonitoringConstant, VpsService) {
+        this.$q = $q;
+        this.$translate = $translate;
+        this.CloudMessage = CloudMessage;
+        this.serviceName = $stateParams.serviceName;
+        this.VpsActionService = VpsActionService;
+        this.VpsMonitoringConstant = VpsMonitoringConstant;
+        this.VpsService = VpsService;
+
+        this.loaders = {
+            init: false
+        };
+        this.data = {};
+        this.period = "LASTDAY";
+    }
+
+    $onInit () {
+        this.loadOptions();
+        this.loadMonitoring();
+    }
+
+    loadMonitoring () {
+        this.loaders.init = true;
+        this.reset();
+        this.VpsService.getMonitoring(this.serviceName, this.period)
+            .then(data => {
+                this.data = data;
+                this.humanizeData(data.cpu.values[0].points, this.monitoring.cpu);
+                this.humanizeData(data.ram.values[0].points, this.monitoring.ram);
+                this.humanizeData(data.netRx.values[0].points, this.monitoring.net[0]);
+                this.humanizeData(data.netTx.values[0].points, this.monitoring.net[1]);
+                this.generateLabels(data.cpu.values[0].points, data.cpu.pointInterval, data.cpu.pointStart, this.monitoring.labels);
+            })
+            .catch(() => {
+                this.CloudMessage.error(this.$translate.instant("vps_configuration_monitoring_fail"));
+                this.error = true;
+            })
+            .finally(() => { this.loaders.init = false });
+    }
+
+    reset () {
+        this.monitoring = {
+            cpu: [],
+            ram: [],
+            net: [[], []],
+            labels: []
+        };
+    }
+
+    humanizeData (data, tab) {
+        _.forEach(data, element => {
+            if (element && element.y) {
+                tab.push(element.y);
+            } else {
+                tab.push(0);
+            }
+        });
+    }
+
+    generateLabels (data, interval, start, tab) {
```

Comment:
To many parameters, bad smell, this is new code...

Reply:
About this, I used to keep consistency with iplb project. Search for "humanize" in iplb.